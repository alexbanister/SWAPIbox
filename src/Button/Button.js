import React from 'react';
import PropTypes from 'prop-types';

const Button = ( { label } ) => {
  return (
    <span className='button'>{label}</span>
  );
};

Button.propTypes = {
  label: PropTypes.string.isRequired
};

export default Button;
