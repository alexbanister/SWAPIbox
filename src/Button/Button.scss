@import "../variables.scss";

.button {
  @include radial-gradient($light-grey, $dark-grey);
  height: 20px;
  padding: 1em 3em;
  margin: 1em;
  border: 1px solid $highlight-dark;
  box-shadow: $shadow;
  color: $background;
  &:hover {
    color: $highlight-dark;
    cursor: pointer;
    box-shadow: none;
  }
}
