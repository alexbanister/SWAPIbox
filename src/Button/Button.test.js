import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from './Button';



describe('Button', () => {
  it('should render correctly and match the snapshot', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<Button onClick={mockFunc} />);

    expect(wrapper).toMatchSnapshot();
  })
})