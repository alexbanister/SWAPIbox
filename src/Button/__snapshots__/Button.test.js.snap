// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
